extensions/objects/templates/svg-filename-editor.component.ts

```diff
@@ -16,1552 +16,1606 @@
-          ctrl.savedSvgDiagram = svgString;
-          resampledFile = (
-            ImageUploadHelperService.convertImageDataToImageFile(
-              svgDataURI));
-          if (
-            ctrl.imageSaveDestination ===
-            IMAGE_SAVE_DESTINATION_LOCAL_STORAGE) {
-            ctrl.saveImageToLocalStorage(dimensions, resampledFile);
-          } else {
-            ctrl.loadingIndicatorIsShown = true;
-            ctrl.postSvgToServer(
-              dimensions, resampledFile).then(function(data) {
-              // Pre-load image before marking the image as saved.
-              var img = new Image();
-              img.onload = function() {
-                ctrl.setSavedSvgFilename(data.filename, false);
-                var dimensions = (
-                  ImagePreloaderService.getDimensionsOfImage(data.filename));
-                ctrl.svgContainerStyle = {
-                  height: dimensions.height + 'px',
-                  width: dimensions.width + 'px'
-                };
-                $scope.$applyAsync();
-                ctrl.loadingIndicatorIsShown = false;
-              };
-              img.src = getTrustedResourceUrlForSvgFileName(data.filename);
-            }, function(parsedResponse) {
-              ctrl.loadingIndicatorIsShown = false;
-              AlertsService.addWarning(
-                parsedResponse.error || 'Error communicating with server.');
-              $scope.$applyAsync();
-            });
-          }
-        }
-      };
-
-      ctrl.isDiagramSaved = function() {
-        return ctrl.diagramStatus === STATUS_SAVED;
-      };
-
-      ctrl.createCustomToSVG = function(toSVG, type, id) {
-        return function() {
-          var svgString = toSVG.call(this);
-          var domParser = new DOMParser();
-          var doc = domParser.parseFromString(svgString, 'image/svg+xml');
-          var parentG = doc.querySelector(type);
-          parentG.setAttribute('id', id);
-          return doc.documentElement.outerHTML;
-        };
-      };
-
-      var loadGroupedObject = function(objId, obj, groupedObjects) {
-        // Checks if the id starts with 'group' to identify whether the
-        // svg objects are grouped together.
-        if (objId.startsWith('group')) {
-          // The objId is of the form "group" + number.
-          const GROUP_ID_PREFIX_LENGTH = 5;
-          var groupId = parseInt(objId.slice(GROUP_ID_PREFIX_LENGTH));
-          // Checks whether the object belongs to an already existing group
-          // or not.
-          if (groupedObjects.length <= groupId) {
-            groupedObjects.push([]);
-          }
-          obj.toSVG = ctrl.createCustomToSVG(
-            obj.toSVG, obj.type, obj.id);
-          groupedObjects[groupId].push(obj);
+      }
+    }
+  }
+
+  isDiagramSaved(): boolean {
+    return this.diagramStatus === this.STATUS_SAVED;
+  }
+
+  createCustomToSVG(
+      toSVG: () => string, selector: string, id: string, ctx: unknown
```

Comment:
```suggestion
    toSVG: () => string, selector: string, id: string, ctx: unknown
```

Reply:
This gives an error. The function arguments that go in a new line are supposed to get two extra indents from the previous lines indent.